main/src/main/java/org/apache/karaf/main/lock/SimpleFileLock.java

```diff
@@ -80,29 +80,12 @@ public void release() throws Exception {
         lock = null;
     }
  
-    public boolean isAlive() throws Exception {
+    public synchronized boolean isAlive() throws Exception {
         return lock != null && lock.isValid() && lockPath.exists();
     }
 
     private static File getKarafLock(File lock,Properties props) {
-        File rc = null;
-
-        String path = lock.getPath();
-        if (path != null) {
-            rc = validateDirectoryExists(path, "Invalid " + PROPERTY_LOCK_DIR + " system property");
-        }
-
-        if (rc == null) {
-            path = props.getProperty(PROP_KARAF_BASE);
-            if (path != null) {
-                rc = validateDirectoryExists(path, "Invalid " + PROP_KARAF_BASE + " property");
-            }
-        }
-
-        if (rc == null) {
-            rc = lock;
-        }
-        return rc;
+        return validateDirectoryExists(lock.getPath(), "Invalid" + PROPERTY_LOCK_DIR + " system property");
```

Comment:
I think the code behaves differently if {{PROPERTY_LOCK_DIR}} is not set.
Previously, it would default to {{KARAF_BASE}}, now it returns null and will certainly cause a NPE in the constructor.

Reply:
Good point, let me change this part and keep the sync on the methods.

Reply:
Thanks !